Gordon360/Services/ScheduleService.cs

```diff
@@ -24,18 +24,25 @@ public ScheduleService(CCTContext context)
         /// Fetch the schedule item whose id and session code is specified by the parameter
         /// </summary>
         /// <param name="username">The AD Username of the student</param>
+        /// <param name="sessionID">The selected session of the student</param>
         /// <returns>StudentScheduleViewModel if found, null if not found</returns>
-        public async Task<IEnumerable<ScheduleViewModel>> GetScheduleStudentAsync(string username)
+        public async Task<IEnumerable<ScheduleViewModel>> GetScheduleStudentAsync(string username, string? sessionID)
         {
             var account = _context.ACCOUNT.FirstOrDefault(x => x.AD_Username == username);
 
             if (account == null)
             {
-                throw new ResourceNotFoundException() { ExceptionMessage = "The Schedule was not found." };
+                throw new ResourceNotFoundException() { ExceptionMessage = "The Account was not found." };
             }
 
-
-            var sessionCode = Helpers.GetCurrentSession(_context);
+            var sessionCode = "";
+            if (sessionID is not null)
+            {
+                sessionCode = sessionID;
+            } else
+            {
+                sessionCode = Helpers.GetCurrentSession(_context);
+            }
```

Comment:
```suggestion
            var sessionCode = sessionID ?? Helpers.GetCurrentSession(_context);
```

`??` is the null coalescence operator which simply means: var sessionCode = `first thing, unless it is null, then second thing`

Reply:
Thank You for the explanation👍